packages/framework/src/client.ts

```diff
@@ -185,45 +176,45 @@ export class Client {
    * @returns mocked data
    */
   private mock(schema: Schema): Record<string, unknown> {
-    return JSONSchemaFaker.generate(schema as JSONSchema7) as Record<string, unknown>;
+    return JSONSchemaFaker.generate(transformSchema(schema) as any) as Record<string, unknown>;
   }
 
-  private validate(
-    data: unknown,
-    validate: Validate,
+  private async validate<T>(
```

Comment:
`validate` is now `async` to support asynchronous validation for heavy schemas, alongside support for fetching from remote schema stores inside of the validator.